lib/collector/api.js

```diff
@@ -70,26 +70,38 @@ function CollectorAPI(agent) {
    * save some GC time?
    */
   this._methods = {}
-  for (const [accessor, name] of [
-    ['preconnect', 'preconnect'],
-    ['connect', 'connect'],
-    ['settings', 'agent_settings'],
-    ['errors', 'error_data'],
-    ['metrics', 'metric_data'],
-    ['traces', 'transaction_sample_data'],
-    ['shutdown', 'shutdown'],
-    ['events', 'analytic_event_data'],
-    ['customEvents', 'custom_event_data'],
-    ['queryData', 'sql_trace_data'],
-    ['errorEvents', 'error_event_data'],
-    ['spanEvents', 'span_event_data'],
-    ['logEvents', 'log_event_data']
+  for (const name of [
+    'preconnect',
+    'connect',
+    'agent_settings',
+    'error_data',
+    'metric_data',
+    'transaction_sample_data',
+    'shutdown',
+    'analytic_event_data',
+    'custom_event_data',
+    'sql_trace_data',
+    'error_event_data',
+    'span_event_data',
+    'log_event_data'
   ]) {
     const method = new RemoteMethod(name, agent, initialEndpoint)
-    this._methods[accessor] = method
+    this._methods[name] = method
   }
 }
 
+CollectorAPI.prototype.send = function send(method, data, callback) {
+  if (!callback) {
+    this._throwCallbackError()
+  }
+  if (!data) {
+    callback(new TypeError(`must pass data for ${method} to send`))
+    return
+  }
+
+  this._sendData(this._methods[method], data, callback)
```

Comment:
For other people reviewing this: the changes in lines 73-86 are tied to `this._methods`. This PR is simplifying that mapping.